src/main/java/mekanism/common/tile/component/SideConfig.java

```diff
@@ -0,0 +1,56 @@
+package mekanism.common.tile.component;
+
+import net.minecraft.util.EnumFacing;
+
+import java.util.Observable;
+import java.util.Observer;
+
+public class SideConfig {
+    private byte[] directions = new byte[EnumFacing.VALUES.length];
+    private Observable observable = new Observable();
```

Comment:
It should not extend Observable. SideConfig is not a specific type of Observable; if Observable were an interface then yes it should implement the interface, stating that one of its features is that it is observable. But to extend Observable means that it is a class which takes Observable and extends it; whereas in this case, we want to use it. (this is the meaning of "prefer composition over inheritance")

A made-up example of extending Observable would be if you wanted a "SingleObservable", like an observable that could only have one observer.

Yes it does appear setChanged needs to be called, good catch.

I'm not stuck on the use of Observable, I just reached for it for prototyping. It might be better to just implement it from scratch so that we could strong-type the Observer and the data that is passed with the change event.